plugin-jdbc-snowflake/src/main/java/io/kestra/plugin/jdbc/snowflake/RSAKeyPairUtils.java

```diff
@@ -0,0 +1,50 @@
+package io.kestra.plugin.jdbc.snowflake;
+
+import org.bouncycastle.asn1.pkcs.PrivateKeyInfo;
+import org.bouncycastle.jce.provider.BouncyCastleProvider;
+import org.bouncycastle.openssl.jcajce.JceOpenSSLPKCS8DecryptorProviderBuilder;
+import org.bouncycastle.operator.InputDecryptorProvider;
+import org.bouncycastle.operator.OperatorCreationException;
+import org.bouncycastle.pkcs.PKCS8EncryptedPrivateKeyInfo;
+import org.bouncycastle.pkcs.PKCSException;
+
+import java.io.IOException;
+import java.security.*;
+import java.security.spec.InvalidKeySpecException;
+import java.security.spec.PKCS8EncodedKeySpec;
+import java.util.Base64;
+import java.util.Optional;
+
+public class RSAKeyPairUtils {
+    public static PrivateKey deserializePrivateKey(String privateKey, Optional<String> privateKeyPassword) {
+        Security.addProvider(new BouncyCastleProvider());
+        var keyBytes = Base64.getDecoder().decode(privateKey);
+        try {
```

Comment:
I'm not fan of try/catch as a "normal path", better is to either add a property to say wether or not it's encrypted, or base on the fact that a password exist so it must be encrypted.